collections-csharp/ImmutableCollectionsInCSharp/ImmutableCollectionsInCSharpTests/ImmutableCollectionsTests.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Xunit;
+
+namespace ImmutableCollectionsInCSharpTests
+{
+    public class ImmutableCollectionsTests
+    {
+        [Fact]
+        public void GivenImmutablStack()
```

Comment:
Sorry to mislead you with the first review but the mandatory part in the test name is When_Then and not the Given part. 
So, could you please change this and all other test methods to WhenXXXX_ThenXXXXX instead of GivenXXXX. 
Thanks